source/behaviors/offClick/offClick.ts

```diff
@@ -29,16 +30,36 @@ export class OffClickDirective implements OnInit, OnDestroy {
 	}
 
 	ngOnInit() {
-		setTimeout(() => {
-			document.addEventListener('click', this.listener);
-		}, 0);
+		if (this.active) {
+			setTimeout(() => {
+				this.addListener();
+			});
+		}
 	}
 
-	ngOnDestroy() {
+	ngOnChanges(changes: SimpleChanges): void {
```

Comment:
I'll chime in with you shouldn't be using a boolean as a tri-state. it's either false or true. null should never be a valid value.